layers.go

```diff
@@ -1757,7 +1758,19 @@ func (r *layerStore) Modified() (bool, error) {
 			return lmodified, err
 		}
 	}
-	return lmodified || mmodified, nil
+
+	// If the layers.json file has been modified manually, then we have to
+	// reload the storage in any case.
+	info, err := os.Stat(r.layerspath())
+	if err != nil && !os.IsNotExist(err) {
+		return false, errors.Wrap(err, "stat layers file")
+	}
+	if info != nil {
+		tmodified = info.ModTime() != r.layerspathModified
+		r.layerspathModified = info.ModTime()
+	}
+
```

Comment:
What happens in the case where the layerspath did not exist?

Reply:
In that case `info == nil` and we stick to the previous behavior (`tmodified` will be false). This is fine because we will load/create the layerspath in any case afterwards.

Reply:
could we add an early return for `if lmodified || mmodified {return true}` so that we avoid a Stat if that condition is already true?

Reply:
Yep, fixed as suggested :+1: